internal/sops/keyservice/server.go

```diff
@@ -86,6 +101,138 @@ func (ks *Server) decryptWithAge(key *keyservice.AgeKey, ciphertext []byte) ([]b
+	reader, enc := utfbom.Skip(bytes.NewReader(contents))
+	switch enc {
+	case utfbom.UTF16LittleEndian:
+		u16 := make([]uint16, (len(contents)/2)-1)
+		err := binary.Read(reader, binary.LittleEndian, &u16)
+		if err != nil {
+			return nil, err
+		}
+		decoded = []byte(string(utf16.Decode(u16)))
+	case utfbom.UTF16BigEndian:
+		u16 := make([]uint16, (len(contents)/2)-1)
+		err := binary.Read(reader, binary.BigEndian, &u16)
+		if err != nil {
+			return nil, err
+		}
+		decoded = []byte(string(utf16.Decode(u16)))
+	default:
+		decoded, err = ioutil.ReadAll(reader)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// unmarshal file
+	authFile := map[string]interface{}{}
+	err = json.Unmarshal(decoded, &authFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := authFile["clientId"]; ok {
+		s.Values["AZURE_CLIENT_ID"] = val.(string)
+	}
+	if val, ok := authFile["clientSecret"]; ok {
+		s.Values["AZURE_CLIENT_SECRET"] = val.(string)
+	}
+	if val, ok := authFile["clientCertificate"]; ok {
+		s.Values["AZURE_CERTIFICATE_PATH"] = val.(string)
+	}
+	if val, ok := authFile["clientCertificatePassword"]; ok {
+		s.Values["AZURE_CERTIFICATE_PASSWORD"] = val.(string)
+	}
+	if val, ok := authFile["subscriptionId"]; ok {
+		s.Values["AZURE_SUBSCRIPTION_ID"] = val.(string)
+	}
+	if val, ok := authFile["tenantId"]; ok {
+		s.Values["AZURE_TENANT_ID"] = val.(string)
+	}
+	if val, ok := authFile["activeDirectoryEndpointUrl"]; ok {
+		s.Values["ActiveDirectoryEndpoint"] = val.(string)
+	}
+	if val, ok := authFile["resourceManagerEndpointUrl"]; ok {
+		s.Values["ResourceManagerEndpoint"] = val.(string)
+	}
+	if val, ok := authFile["activeDirectoryGraphResourceId"]; ok {
+		s.Values["GraphResourceID"] = val.(string)
+	}
+	if val, ok := authFile["sqlManagementEndpointUrl"]; ok {
+		s.Values["SQLManagementEndpoint"] = val.(string)
+	}
+	if val, ok := authFile["galleryEndpointUrl"]; ok {
+		s.Values["GalleryEndpoint"] = val.(string)
+	}
+	if val, ok := authFile["managementEndpointUrl"]; ok {
+		s.Values["ManagementEndpoint"] = val.(string)
+	}
+
+	resource := azure.PublicCloud.ResourceIdentifiers.KeyVault
+	if a, err := s.ClientCredentialsAuthorizerWithResource(resource); err == nil {
+		return a, err
+	}
+	if a, err := s.ClientCertificateAuthorizerWithResource(resource); err == nil {
+		return a, err
+	}
+	return nil, errors.New("auth file missing client and certificate credentials")
+}
+
+func (ks *Server) encryptWithAzureKeyvault(key *keyservice.AzureKeyVaultKey, plaintext []byte) ([]byte, error) {
+	var err error
+
+	kv := keyvault.New()
+	if kv.Authorizer, err = ks.newKeyvaultAuthorizerFromFile(filepath.Join(ks.HomeDir, "azure_kv.json")); err != nil {
+		return nil, err
+	}
+
+	plainstring := string(plaintext)
+	result, err := kv.Encrypt(context.Background(), key.VaultUrl, key.Name, "", keyvault.KeyOperationsParameters{Algorithm: keyvault.RSAOAEP256, Value: &plainstring})
+	if err != nil {
+		return nil, fmt.Errorf("Failed to encrypt data: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to encrypt data: %w", err)
```

(applies to all errors)